component/src/main/java/io/siddhi/extension/execution/string/FillTemplateFunctionExtension.java

```diff
@@ -118,23 +124,45 @@
     @Override
     protected Object execute(Object[] objects, State state) {
         String sourceString = (String) objects[0];
+        Map<String, Object> valueMap = null;
         Matcher templateMatcher = templatePattern.matcher(sourceString);
         String match;
         Matcher indexMatcher;
-        int index;
-        while (templateMatcher.find()) {
-            match = templateMatcher.group(0);
-            indexMatcher = indexPattern.matcher(match);
-            if (indexMatcher.find()) {
-                index = Integer.parseInt(indexMatcher.group(0));
-                if (index < 1 || index >= objects.length) {
-                    throw new SiddhiAppRuntimeException("Index given for template elements "
-                            + "should be greater than 0 and less than '" + objects.length + "'. But found "
-                            + index + " in" + " the template '" + sourceString + "'.");
+        if (objects[1] instanceof HashMap) {
```

Comment:
do a instanceOf map rather than Hashmap  and cast to Map as well

Reply:
We know whether its str:fillTemplate(STRING, STRING|INT|LONG|DOUBL|BOOL, ...) or str:fillTemplate(STRING, OBJECT) at the init itself, therefore, you do not need to check for instance of to decide the which type you are handling here. But within the str:fillTemplate(STRING, OBJECT) case you have to check if the OBJECT is an instance of Map as @pcnfernando  mentioned.